index.js

```diff
@@ -162,6 +164,58 @@ const inventoryCommand = (message) => {
   });
 };
 
+const lookUpCommand = (message) => {
+  const originalMessage = message.content;
+  const fuzzySearchKey = originalMessage.split(" ").splice(1).join(" ");
+  const fuzzySearchUrl =
+    "https://db.ygoprodeck.com/api/v7/cardinfo.php?&fname=";
+
+  console.log(fuzzySearchUrl + fuzzySearchKey);
+
+  axios
+    .get(fuzzySearchUrl + fuzzySearchKey)
+    .then((cardSearchResults) => {
+      const objectsReturnedBySearch = cardSearchResults.data.data;
+      const exactMatchObject = objectsReturnedBySearch.find(
+        (element) => element.name.toLowerCase() === fuzzySearchKey.toLowerCase()
+      );
+
+      const closestMatchObject = objectsReturnedBySearch.find((element) =>
+        element.name.toLowerCase().startsWith(fuzzySearchKey.toLowerCase())
+      );
+
+      const bestCardResult =
+        exactMatchObject || closestMatchObject || objectsReturnedBySearch[0];
+
+      const attachment = new Discord.MessageAttachment(
+        bestCardResult.card_images[0].image_url
+      );
+      message.reply(
+        "The top result for your search is: " + bestCardResult.name
+      );
+      message.channel.send(attachment);
+
+      const suggestedSearches = objectsReturnedBySearch
+        .filter((item) => item.name != bestCardResult.name)
```

Comment:
generally if you're given an `id`, it's best to compare `id`s since they're guaranteed to be unique. We can be reasonably sure that 2 cards probably wont have the same name, but it's a good habit to build to check ids when possible.